### PR TITLE
Invalidate scroll on arranging instead of invalidating on measuring

### LINKED ISSUE #5651: Changing Selected TabItem resets the scroll position of a ListBox
**Describe the bug**
This is definitely a weird one. I was able to simplify down to a minimum example but I still don't see what the issue is. I'm not sure if it's how I'm doing things or if it's a bug on Avalonia's side.

Basically, in my setup, if you scroll the listbox then change tabs - the listbox scroll position gets reset. `ListBox.LayoutUpdated` is getting called so I think that's part of it (not actually that the scroll position is being reset) but the question is still: *why?*

Screen recording:

**To Reproduce**
Here's my minimum reproducible example:

MainWindow.axaml:
```xml
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" Width="600" Height="450"
        x:Class="AvaloniaTabChangeResettingControls.MainWindow">
  <Grid RowDefinitions="*,Auto" ColumnDefinitions="Auto,*">
    <ListBox Width="200" Items="{Binding ListBoxItems}"/>
    <TabControl Grid.Column="1">
      <TabItem Header="Tab1">
        Tab1&#10;Content
      </TabItem>
      <TabItem Header="Tab2">
        Tab2 Content
      </TabItem>
    </TabControl>
    <Grid Grid.Row="1" Grid.ColumnSpan="2"/>
  </Grid>
</Window>
```

MainWindow.xaml.cs:
```csharp
using System.Collections.Generic;
using Avalonia;
using Avalonia.Controls;
using Avalonia.Markup.Xaml;

namespace AvaloniaTabChangeResettingControls
{
    public class MainWindow : Window
    {
        public MainWindow()
        {
            AvaloniaXamlLoader.Load(this);
            this.DataContext = this;
        }

        public List<string> ListBoxItems { get; } = new List<string>
        {
            "Item1",
            "Item2",
            "Item3",
            "Item4",
            "Item5",
            "Item6",
            "Item7",
            "Item8",
            "Item9",
            "Item10",
            "Item11",
            "Item12",
            "Item13",
            "Item14",
            "Item15",
        };
    }
}
```

Some things that oddly seem to be essential to reproducing the error:
- The fact that the contents of each tab item are different sizes (notice the `&#10;` newline character in Tab1. If you remove that, then the issue does not happen)
- The 2nd, inner Grid with no content (if I remove that, the issue does not happen - but no idea why)

*I do not think it matters how `ListBox.Items` is set (binding vs codebehind using FindControl)*

**Expected behavior**
I would expect the ListBox to not update it's layout (change scroll position) when tab selection is changed.

**Desktop (please complete the following information):**

- OS: Windows 10
- Version 10.0.0

### PR DESCRIPTION
**Pull request by maxkatz6**
## What is the current behavior?
Scroll is invalidated on measuring, which theoretically can be invoked multiple times during single pass. Grid control actually does invoke measuring more than once.

## What is the updated/expected behavior with this PR?
Scroll is invalidated only on arranging, when we have final size and can compare it with previous size.

## Fixed issues
Fixes https://github.com/AvaloniaUI/Avalonia/issues/5651

### CODE REVIEW

**Comment by maxkatz6** [comment]
If this change does make sense, I can write some tests for it.
I am just not sure if I understand expected behavior correctly.

P.S. it's the first significant change in this file for years, whoa.

**Review by grokys — APPROVED** [review]
Makes sense and I don't think should cause any problems (famous last words).

Yeah this code is _old_ and hopefully going away soon! So not sure tests are really necessary.

**Comment by maxkatz6** [comment]
Closing this, because my fix has bigger impact on current functionality, that I can handle with my understanding of this code.
It's specifically about tests as well:
- VirtualizingStackPanelTests are fine, some of them could be removed, some of them can be updated, to follow new behavior.
- ItemsPresenterTests_Virtualization on other hand mostly should just work even with my fix, but they don't.

I assume it's because current logic of items presenter measuring depends on materialized items, what was called with CreateAndRemoveContainers from UpdateControls previously.

FYI @grokys

File: src/Avalonia.Controls/VirtualizingStackPanel.cs [hunk 1]
```diff
@@ -84,14 +84,18 @@ protected override Size MeasureOverride(Size availableSize)
...
             }
 
             return base.MeasureOverride(availableSize);
         }
 
         protected override Size ArrangeOverride(Size finalSize)
         {
+            if (_availableSpace != finalSize)
+            {
+                Controller?.UpdateControls();
```
Comment (maxkatz6) [inline src/Avalonia.Controls/VirtualizingStackPanel.cs:96]: Btw, UpdateControls was already called in ArrangeOverride couple of lines below.

**Pull request closed without merging**